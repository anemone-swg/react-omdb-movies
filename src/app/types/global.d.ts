declare namespace NodeJS {
  interface ProcessEnv {
    NODE_ENV: "development" | "production";
    OMDB_API_KEY: string;
  }
}

declare module "*.scss";

declare module "*.module.scss" {
  const classes: { [key: string]: string };
  export default classes;
}

declare module "*.jpg" {
  const value: string;
  export default value;
}

// для того чтобы убрать ошибки при переносе stylelint конфига на TS
// но смысла переносить stylelint на TS нет, т.к. он написан на JS
declare module "stylelint-config-recess-order";
declare module "stylelint-config-standard-scss";
