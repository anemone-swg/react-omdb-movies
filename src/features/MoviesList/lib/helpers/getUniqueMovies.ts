import type { Movie } from "../../model/types/movie";

export const getUniqueMovies = (data: Movie[]) =>
  data.filter(
    (movie, index, self) =>
      index === self.findIndex((m) => m.imdbID === movie.imdbID),
  );
