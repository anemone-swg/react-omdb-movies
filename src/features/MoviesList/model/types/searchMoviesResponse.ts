import type { Movie } from "./movie.ts";

export interface SearchMoviesResponse {
  Search: Movie[];
  totalResults: string;
  Response: "True" | "False";
  Error?: string;
}
