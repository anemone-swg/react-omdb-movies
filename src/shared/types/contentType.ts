export type contentType = "movie" | "series" | "episode" | undefined;
