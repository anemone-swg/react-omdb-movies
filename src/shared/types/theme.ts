/**
 * Тип темы приложения.
 * - `"light"` - светлая тема
 * - `"dark"` - темная тема
 */
export type Theme = "light" | "dark";
