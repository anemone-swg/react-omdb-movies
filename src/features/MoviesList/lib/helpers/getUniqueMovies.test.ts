import { getUniqueMovies } from "./getUniqueMovies";
import { Movie } from "../../model/types/movie";

describe("getUniqueMovies", () => {
  const mockMovies1: Movie[] = [
    {
      imdbID: "1",
      Title: "Movie 1",
      Year: "2020",
      Type: "movie",
      Poster: "N/A",
    },
    {
      imdbID: "1",
      Title: "Movie 1 duplicate",
      Year: "2020",
      Type: "movie",
      Poster: "N/A",
    },
    {
      imdbID: "2",
      Title: "Movie 2",
      Year: "2021",
      Type: "series",
      Poster: "N/A",
    },
  ];

  const mockMovies2: Movie[] = [
    {
      imdbID: "1",
      Title: "Movie 1",
      Year: "2020",
      Type: "movie",
      Poster: "N/A",
    },
    {
      imdbID: "2",
      Title: "Movie 2",
      Year: "2021",
      Type: "series",
      Poster: "N/A",
    },
  ];

  const mockMovies3: Movie[] = [
    {
      imdbID: "1",
      Title: "Movie 1",
      Year: "2020",
      Type: "movie",
      Poster: "N/A",
    },
    {
      imdbID: "1",
      Title: "Movie 1 duplicate",
      Year: "2020",
      Type: "movie",
      Poster: "N/A",
    },
    {
      imdbID: "1",
      Title: "Movie 2",
      Year: "2021",
      Type: "series",
      Poster: "N/A",
    },
  ];

  test("уберет дубликаты", () => {
    // expect(getUniqueMovies(mockMovies)).toEqual([
    //   {
    //     imdbID: "1",
    //     Title: "Movie 1",
    //     Year: "2020",
    //     Type: "movie",
    //     Poster: "N/A",
    //   },
    //   {
    //     imdbID: "2",
    //     Title: "Movie 2",
    //     Year: "2021",
    //     Type: "series",
    //     Poster: "N/A",
    //   },
    // ]);
    expect(getUniqueMovies(mockMovies1)).toMatchSnapshot();
  });
  test("нет дубликатов", () => {
    expect(getUniqueMovies(mockMovies2)).toEqual(mockMovies2);
  });
  test("все дубликаты", () => {
    expect(getUniqueMovies(mockMovies3)).toEqual([mockMovies3[0]]);
  });
});
