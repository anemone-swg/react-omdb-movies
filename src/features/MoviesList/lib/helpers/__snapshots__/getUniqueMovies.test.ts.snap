// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`getUniqueMovies уберет дубликаты 1`] = `
[
  {
    "Poster": "N/A",
    "Title": "Movie 1",
    "Type": "movie",
    "Year": "2020",
    "imdbID": "1",
  },
  {
    "Poster": "N/A",
    "Title": "Movie 2",
    "Type": "series",
    "Year": "2021",
    "imdbID": "2",
  },
]
`;
